### PR TITLE
Upgrade celery to 4.2

### PR DESCRIPTION
**Pull request by sfdye**
Closes #1249.

According to the official [upgrading guide](http://docs.celeryproject.org/en/latest/whatsnew-4.0.html#step-4-upgrade-to-celery-4-0
) from 3.X to 4.X, the only thing that could break the existing code for `cookicutter-django` seems to be the new lower case setting names. However this change is backward-compatible, so until celery officially depreciates upper case settings name, we are good.

Therefore this update would be very simple, just one incompatible settings name change. I've already bumped celery from 3.1.25 to 4.1 for one of my company's projects in production and so far I don't see any issues arise from it.

@luzfcb I haven't encounter the timezone issues you mentioned, so if you can reproduce it please let me know.

### CODE REVIEW

**Comment by browniebroke** [comment]
We've started a project soon after the Celery 4 release so we started with it. It's been a while we've been using it without any issue apart from [a timezone issue](https://github.com/celery/celery/issues/4160), probably the same as @luzfcb. 

It's been fixed in [this pull request](https://github.com/celery/celery/pull/4173), but it's yet to be released, the Celery project seems a bit stretched in terms of resources at the moment...

**Comment by webyneter** [comment]
@browniebroke I guess, we can wait then :)

File: {{cookiecutter.project_slug}}/config/settings/local.py [hunk 1]
```diff
@@ -81,7 +81,7 @@
 {% if cookiecutter.use_celery == 'y' %}
 ########## CELERY
 # In development, all tasks will be executed locally by blocking until the task returns
-CELERY_ALWAYS_EAGER = True
+CELERY_TASK_ALWAYS_EAGER = True
```
Comment (foarsitter) [inline {{cookiecutter.project_slug}}/config/settings/local.py]: I suggest setting [CELERY_EAGER_PROPAGATES](http://docs.celeryproject.org/en/latest/userguide/configuration.html#task-eager-propagates)=True here when using task_always_eager in local development so the exceptions will not be suppressed.

File: {{cookiecutter.project_slug}}/requirements/base.txt [hunk 2]
```diff
@@ -50,8 +50,7 @@ django-redis==4.8.0
 redis>=2.10.5
 
 {% if cookiecutter.use_celery == "y" %}
-celery==3.1.25 # pyup: <4.0
-{% endif %}
+celery==4.1.0 # pyup: <5.0
```
Comment (browniebroke) [inline {{cookiecutter.project_slug}}/requirements/base.txt]: Apologies, I've messed up your diff while resolving a conflict. You'll need to re-add the `{% endif %}`.

Comment (sfdye) [reply {{cookiecutter.project_slug}}/requirements/base.txt]: Sure, no worres.

File: {{cookiecutter.project_slug}}/config/settings/local.py (hunk 1 above)
Comment (sfdye) [reply {{cookiecutter.project_slug}}/config/settings/local.py]: Thanks for the advice, I think it makes sense.

**Comment by jayfk** [comment]
Can someone confirm that this works on production? I had various problem with this, celery swallowing errors and not sending them to sentry was one of them.

**Comment by webyneter** [comment]
@sfdye did you have a chance to test this one out in production already?

**Comment by sfdye** [comment]
@webyneter @jayfk I am already running this in production, with no issues. Maybe my use case is not complex enough to reproduce the same error? Would still want to wait for celery 4.2 to roll out though.

**Comment by browniebroke** [comment]
I'm using Celery 4 in production as well, I haven't got any Celery specific issues, but our use case is pretty basic too. The [4.2 milestone](https://github.com/celery/celery/milestone/19) is nearly completed on the Celery project.

**Comment by sfdye** [comment]
@webyneter Would you mind helping me with the conflicts? I am not too sure what you changed in that huge PR 😝

**Comment by orzarchi** [comment]
Any news?

**Comment by webyneter** [comment]
Haven't got any issues using Celery 4.1 on one of my recent projects in production either.

**Comment by sfdye** [comment]
Still waiting for the upstream to release a stable version 4.2

**Comment by webyneter** [comment]
@sfdye yes we do :)

**Comment by webyneter** [comment]
I guess, we won't see 4.2 soon, at least not until the [end of the coming week](https://github.com/celery/celery/issues/4387)...

**Comment by sfdye** [comment]
It's happening. Updated this branch as well.

**Comment by browniebroke** [comment]
@sfdye do you have anything to add on this? There is another PR #1679 which is depending on this.

**Comment by sfdye** [comment]
This PR simply upgrades the major version. Maybe leave the conf tweaks to the other PR?

**Comment by browniebroke** [comment]
Yes agreed, I also think it's ready for merge.

**Comment by webyneter** [comment]
@sfdye please remove the `CELERY_TASK_EAGER_PROPAGATES = True` line before merging -- I've made the [same changes to a sibling PR](https://github.com/pydanny/cookiecutter-django/pull/1679/files#diff-9966510e853c64f6dc65446359c466c0R82).

**Comment by webyneter** [comment]
@sfdye and `CELERY_ALWAYS_EAGER` -> `CELERY_TASK_ALWAYS_EAGER` [too](https://github.com/pydanny/cookiecutter-django/pull/1679/files#diff-9966510e853c64f6dc65446359c466c0R80).

**Comment by sfdye** [comment]
Done

**Comment by sfdye** [comment]
What are we still waiting for?

**Comment by webyneter** [comment]
@sfdye for the author of the PR to merge it into :)

**Comment by sfdye** [comment]
@webyneter I see 😂

**Pull request merged**